src/registries/figure_registry.ts

```diff
@@ -58,6 +61,40 @@ function getChartMenu(
     },
     getCopyMenuItem(figureId, env),
     getCutMenuItem(figureId, env),
+    {
+      id: "copy_as_image",
+      name: _t("Copy as image"),
+      icon: "o-spreadsheet-Icon.COPY_AS_IMAGE",
+      sequence: 4,
+      execute: async () => {
+        const figureSheetId = env.model.getters.getFigureSheetId(figureId)!;
+        const figure = env.model.getters.getFigure(figureSheetId, figureId)!;
+        const chartType = env.model.getters.getChartType(figureId);
+        const runtime = env.model.getters.getChartRuntime(figureId);
+        const imageUrl = chartToImage(runtime, figure, chartType)!;
+        const innerHTML = `<img src="${xmlEscape(imageUrl)}" />`;
+        const blob = await chartToImageFile(runtime, figure, chartType)!;
+
+        env.clipboard.write({
+          "text/html": innerHTML,
+          "image/png": blob || undefined,
```

Comment:
is the || undefined useful ? chartToImageFile is typed to return File | undefined already